docs/source/main_classes/trainer.mdx

```diff
@@ -189,6 +189,30 @@ that make things deterministic (.e.g., `torch.backends.cudnn.deterministic`) may
 can't be done by default, but you can enable those yourself if needed.
 
 
+## Specific GPUs Selection
+
+If you have multiple GPUs and you'd like to use only 1 or a few of those GPUs, set the environment variable `CUDA_VISIBLE_DEVICES`. For example to run only on GPUs 0 and 2, you can do:
+
+```bash
+CUDA_VISIBLE_DEVICES=0,2 python trainer-program.py ...
+```
+
+So now pytorch will see only 2 gpus where your real GPU 0 becomes gpu0, and your real GPU 2 becomes gpu1.
```

Comment:
Should we add here a mention on how to do the same with a distributed launch? Data parallel doesn't work for all models and isn't recommended by PyTorch.

Reply:
it's exactly the same for any DP/DDP/no-DP mode (well with the exception of the `deepspeed` launcher where it depends on the args being passed).

Reply:
yup, the same with DDP, reference: https://discuss.pytorch.org/t/specify-which-gpus-to-use-with-torch-distributed-launch/115025/2?u=stas

Reply:
should I edit gpu0, gpu1 to  cuda:0, cuda:1 as in the answer I linked to?

Reply:
And let me know if I should explicitly say that the same applies to DP/DDP/no-DP

Reply:
Yes, I know it is the same, but it will be clearer to the user to show them you can do
```bash
CUDA_VISIBLE_DEVICES=0,2 python -m torch.distributed.launch trainer-program.py
```

Reply:
Makes total sense, Sylvain.

It's dangerous to ask me to write more - now this section is 10x larger. :)

Reply:
Ah ah, will do!